pull_request_template.md

```diff
@@ -0,0 +1,118 @@
+## PR Writing Checklist
+
+If your PR involves adding or updating writing content,
+please check the following items before submitting your PR for review:
+
+- [ ] **Avoid [inline links].
+  Instead, use [reference links] with a list of
+  [link reference definitions] at the bottom of your page.**
+
+### Example:
+
+Not recommended:
+
+```md
+For more information, consult
+[reference links](https://github.github.com/gfm#reference-links)
+for [GitHub Flavored Markdown](https://github.github.com/gfm).
+
+```
+
+Recommended:
+
+```md
+For more information, consult [reference links] for [GitHub Flavored Markdown].
+
+(Include this list of link reference definitions at the bottom of the markdown file)
+
+[reference links]: https://github.github.com/gfm/#reference-link
+[GitHub Flavored Markdown]: https://github.github.com/gfm/
+```
+
+- [ ] **Avoid duplicating a referenced link definition by using
```

Comment:
Sure, though I'm not sure this is important enough to warrant this list. And, actually, this reminds me of another rule we try to follow (though not always). We generally link to, say, a class spec page ONCE. After that, we don't link to it again, unless it's much later in the page.  @kwalrath, please weigh in.

Reply:
TLDR: IMHO this happens, and it will continue happening, so we can spare a checklist item for some guidance on it. 

> I'm not sure this is important enough to warrant this list.

Fair enough, happy to take this out if it's not high up on our list.

> We generally link to, say, a class spec page ONCE. After that, we don't link to it again, unless it's much later in the page.

 It's worth noting that we do often reference the same link multiple times within a page. That may be something that we want to avoid doing, or it may be something that we simply can't avoid.  Here is a short sampling from some highly-trafficked pages:

From editor.md : 
```
Alternatively, you can also use IntelliJ:

* [IntelliJ IDEA Community](https://www.jetbrains.com/idea/download/), version 2017.1 or later
* [IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/download/), version 2017.1 or later
```

From deployment/ios :
```
Flutter app to the [App Store][appstore] and [TestFlight][testflight].
...
## Release your app on TestFlight

[TestFlight][testflight] allows developers to push their apps


 After the archive has been successfully validated, click
   **Upload to App Store...**. You can follow the status of your
   build in the Activities tab of your app's details page on
   [App Store Connect][appstoreconnect_login].


1. Navigate to the TestFlight tab of your app's application
   details page on [App Store Connect][appstoreconnect_login].
```

From flutter-for/android-devs.md:

```

Plugins are missing, you can build your own following the
[developing packages and plugins](/docs/development/packages-and-plugins/developing-packages) page.

...
To learn how to build a native platform integration,
see [Developing packages and plugins][].

```
I'm happy to provide a larger sample if needed, but my suspicion is that this is something that will happen regularly, and encouraging contributors to use labels could help keep things tidy.

Reply:
Yeah, I know we do. It's one of those things you have to balance the usefulness of it. We are definitely not consistent.